plugin/src/main/java/org/opensearch/ml/action/model_group/DeleteModelGroupTransportAction.java

```diff
@@ -81,42 +104,107 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<Delete
 
                     SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().query(query);
                     SearchRequest searchRequest = new SearchRequest(ML_MODEL_INDEX).source(searchSourceBuilder);
-                    client.search(searchRequest, ActionListener.wrap(mlModels -> {
-                        if (mlModels == null || mlModels.getHits().getTotalHits() == null || mlModels.getHits().getTotalHits().value == 0) {
-                            deleteModelGroup(deleteRequest, modelGroupId, wrappedListener);
-                        } else {
-                            throw new MLValidationException("Cannot delete the model group when it has associated model versions");
-                        }
-
-                    }, e -> {
-                        if (e instanceof IndexNotFoundException) {
-                            deleteModelGroup(deleteRequest, modelGroupId, wrappedListener);
-                        } else {
-                            log.error("Failed to search models with the specified Model Group Id " + modelGroupId, e);
-                            wrappedListener.onFailure(e);
-                        }
-                    }));
+
+                    SearchDataObjectRequest searchDataObjectRequest = new SearchDataObjectRequest.Builder()
+                        .indices(ML_MODEL_INDEX)
+                        .searchSourceBuilder(searchSourceBuilder)
+                        .build();
+
+                    sdkClient
+                        .searchDataObjectAsync(searchDataObjectRequest, client.threadPool().executor(GENERAL_THREAD_POOL))
+                        .whenComplete((sr, st) -> {
+                            if (sr != null) {
+                                try {
+                                    SearchResponse searchResponse = SearchResponse.fromXContent(sr.parser());
+                                    SearchHit[] searchHits = searchResponse.getHits().getHits();
+                                    if (searchHits.length == 0) {
+                                        deleteModelGroup(deleteRequest, tenantId, wrappedListener);
+                                    } else {
+                                        actionListener
+                                            .onFailure(
+                                                new OpenSearchStatusException(
+                                                    "Cannot delete the model group when it has associated model versions",
+                                                    RestStatus.CONFLICT
```

Comment:
It doesn't need to be. I feel like this is more relevant exception than we had before. Thoughts?

Reply:
Per my comments elsewhere in review, I'd personally prefer to keep existing exceptions/code logic as-is if we're just migrating the wrappers.  I agree it may be a better exception type but would prefer that in a separate PR (perhaps on the main branch backported here).